### PR TITLE
Add HTTP routing to master for Azure

### PR DESCRIPTION
**Pull request by mellenburg**
Currently, the acs template does not provide the user with a route to the master for DC/OS UI usage. This was due to security requirements and previous issues with oauth. Now that we can oauth in the acs templates without issue, this patch will add routes for http/https when oauth is *available* and then will only add extra network security rules when oauthEnabled=true at deploy time.

Thus, with this PR, when someone deploys a oauthEnabled=false cluster, the traffic will not be routed to the master, but it can easily be turned on by opening the Network Security Group rule. Additionally, if a downstream DC/OS variant is built and if it uses `custom_auth`  the routes will be enabled.

## Checklist for all PR's

  - [x] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

### CODE REVIEW

**Comment by mellenburg** [comment]
@lingmann PTAL This is a little uglier than I wanted, but then again, templates like this weren't designed for conditionals in mind and only really support it with nested templates. I get around that by being a little tricky with maps, but it allows us to deploy different resources dynamically

**Comment by mellenburg** [comment]
@mesosphere-mergebot bump-ee

**Comment by mellenburg** [comment]
@mesosphere-mergebot bump-ee

**Comment by mellenburg** [comment]
@mesosphere-mergebot bump-ee

**Comment by mellenburg** [comment]
@mesosphere-mergebot bump-ee

**Comment by mellenburg** [comment]
@mesosphere-mergebot bump-ee

**Comment by mellenburg** [comment]
@mesosphere-mergebot bump-ee

**Comment by mellenburg** [comment]
@mesosphere-mergebot bump-ee

**Comment by mellenburg** [comment]
@mesosphere-mergebot  bump-ee

**Comment by mellenburg** [comment]
@mesosphere-mergebot bump-ee

**Comment by mellenburg** [comment]
@mesosphere-mergebot bump-ee

**Comment by mellenburg** [comment]
Template artifacts produced in https://github.com/mesosphere/dcos-enterprise/pull/748 were verified to work so this PR is safe for downstream consumption

**Comment by mellenburg** [comment]
@mesosphere-mergebot bump-ee

**Comment by mellenburg** [comment]
@mesosphere-mergebot bump-ee

File: gen/build_deploy/azure.py [hunk 1]
```diff
@@ -43,13 +43,21 @@ def validate_provider(provider):
     assert provider == 'azure'
 
 
+def calculate_public_route_to_master_available(oauth_available, custom_auth):
+    if oauth_available == 'true' or custom_auth == 'true':
+        return 'true'
+    else:
+        return 'false'
```
Comment (mellenburg) [inline gen/build_deploy/azure.py]: Its more that if you decide to roll your own, you can just set your config preferences to be oauth_availble = "false" in which case the generated template will have no auth in DC/OS and an HTTP/S route to master. There is no guarantee that "custom_auth" is true if oauth is unavailable

**Comment by mellenburg** [comment]
@mesosphere-mergebot bump-ee

**Comment by mellenburg** [comment]
@lingmann Can we move this forward? I have manually verified downstream. Still looking at a 100% success rate for launching with this template, so I am not worried about flakes

File: gen/azure/templates/acs.json [hunk 2]
```diff
@@ -789,12 +802,96 @@
...
+      "dependsOn": [
+        "[variables('masterLbID')]"
+      ],
+      "properties": {
+        "backendPort": 443,
+        "enableFloatingIP": false,
+        "frontendIPConfiguration": {
+          "id": "[variables('masterLbIPConfigID')]"
+        },
+        "frontendPort": "443",
```
Comment (orsenthil) [inline gen/azure/templates/acs.json]: Let's have `int` type here. Otherwise it reads inconsistent while reviewing.

**Pull request merged**